### ISSUE TITLE
WiiGee Gesture Training/Recognition port for the leap

### ISSUE THREAD

**Issue by schwittlick** [openleap/LeapGestures#1]
i've been working on porting the same gesture recognition and training algorithm you're using to the leap. it looks like you're using this: http://www.wiigee.org/. how is your progress, from what i can see its not yet done, isnt it?  are you still developing this? i started but unfortunately i'm a little too busy to make it usable with the leap at the moment, but i will continue to it asap. 

there's a little longer essay about this (in german only though): https://dl.dropbox.com/u/26339415/poppinga_wiimote_gestenerkennung.pdf

would you be interested in collaborating on this in the future? what do you think about a java version? there's gonna be some sort of gesture recognition included in one of the next sdk releases, would it actually be worth the effort to make this happen if they include it sooner or later anyway? let me know your thoughts.

btw: i talked to benjamin poppinga about license agreements and they're willing to change licensing if neccessary.

marcel

**Comment by Tesserex** [comment]
I'm holding off until we find out what the Leap SDK provides for us in the next releases. Nobody on their end has responded to my questions for clarification about the intended features.

If we do need to make something like this ourselves, I'd sure like the help. A java version is fine with me too but I'm no java expert so it would be up to you. My version so far is running, but doesn't work all too well. I don't know enough about the algorithm to figure out how to improve its performance.

**Comment by schwittlick** [comment]
yeah, its weird. can you link me to the thread where you asked about this please.

did you write this from scratch? since the original implementation is in java i though you took it as the base and ported it to C#. the reason why it might not work properly yet is because it was designed to work with the wiimote. i need to digg deeper as well, but ill pause this until the native leap sdk gesture support turns out to be bad. hope it doesnt ;)

**Comment by Cervator** [comment]
4 basic gestures were released in the updated SDK, which is both good and bad since more could be coming leaving less of a need for libraries yet .. there are still only four gestures and more mystery stuff coming in the future :-)

**Comment by owenversteeg** [comment]
I haven't had time to play with the updated SDK yet but the gesture support looks promising. I especially like that the events give you lots of data, rather than the none-to-minimal data I expected such gestures to return.

I wonder if there are any event handler-type things. I guess I'll find out later this week when I have some time :)
